packages/core/src/node/synchronizer.ts

```diff
@@ -332,6 +342,90 @@ export class Synchronizer extends EventEmitter {
       })
   }
 
+  async listenDepositUtxos(
+    addresses: ZkAddress[],
+    cb?: (utxo: UtxoSql) => void,
+  ) {
+    const lastDeposits = await this.db.findMany('Utxo', {
+      where: {},
+      orderBy: { depositedAt: 'desc' },
+      limit: 1,
+    })
+    const fromBlock = lastDeposits[0]?.blockNumber || 0
+    logger.info('found my deposit from block', fromBlock)
+    this.depositUtxoSubscriber = this.l1Contract.user.events
+      .DepositUtxo({
+        fromBlock,
+        filter: {
+          spendingPubKey: addresses.map(address => address.spendingPubKey()),
+        },
+      })
+      .on('connected', subId => {
+        logger.info(
+          `synchronizer.js: DepositUtxo listener is connected. Id: ${subId}`,
+        )
+      })
+      .on('data', async event => {
+        const { returnValues, blockNumber } = event
+        const owner = addresses.find(addr =>
+          Fp.from(returnValues.spendingPubKey).eq(addr.spendingPubKey()),
+        )
+        if (!owner) throw Error('Failed to find address')
```

Comment:
@wanseob This is throwing a scary error on the coordinator. This shouldn't be an error, it should be okay if the node doesn't have the spending key (it's someone else's deposit in that case).

Reply:
yeah you right, let me fix this

Reply:
Commit 173d222869abc14eaf95130067f598bcdec4ea97 resolves this